src/init/features/functions/index.ts

```diff
@@ -52,11 +148,15 @@ export async function doSetup(setup: any, config: any, options: Options) {
     default: "javascript",
     choices,
   });
-  _.set(setup, "config.functions.ignore", [
-    "node_modules",
-    ".git",
-    "firebase-debug.log",
-    "firebase-debug.*.log",
-  ]);
+  const cbconfig = configForCodebase(setup.config.functions, setup.functions.codebase);
+  switch (language) {
+    case "javascript":
+      cbconfig.ignore = ["node_modules", ".git", "firebase-debug.log", "firebase-debug.*.log"];
+      break;
+    case "typescript":
+      cbconfig.ignore = ["node_modules", ".git", "firebase-debug.log", "firebase-debug.*.log"];
+      break;
+    // add other cases as more languages are supported
```

Comment:
```suggestion
```

(I think this comment is unnecessary! The code makes it pretty obvious what needs to be done to support more languages.)